sympy/calculus/tests/test_util.py

```diff
@@ -63,6 +63,52 @@ def test_not_empty_in():
         Union(Interval(-2, -1, True, False), Interval(2, oo))
 
 
+def test_periodicity():
+    x = Symbol('x')
+    y = Symbol('y')
+
+    assert periodicity(sin(2*x), x) == pi
+    assert periodicity((-2)*tan(4*x), x) == pi/4
+    assert periodicity(sin(x)**2, x) == 2*pi
+    assert periodicity(3**tan(3*x), x) == pi/3
+    assert periodicity(tan(x)*cos(x), x) == 2*pi
+    assert periodicity(sin(x)**(tan(x)), x) == 2*pi
+    assert periodicity(tan(x)*sec(x), x) == 2*pi
+    assert periodicity(sin(2*x)*cos(2*x) - y, x) == pi/2
+    assert periodicity(tan(x) + cot(x), x) == pi
+    assert periodicity(sin(x) - cos(2*x), x) == 2*pi
+    assert periodicity(sin(x) - 1, x) == 2*pi
+    assert periodicity(sin(4*x) + sin(x)*cos(x), x) == pi
+    assert periodicity(exp(sin(x)), x) == 2*pi
+    assert periodicity(log(cot(2*x)) - sin(cos(2*x)), x) == pi
+    assert periodicity(sin(2*x)*exp(tan(x) - csc(2*x)), x) == pi
+    assert periodicity(cos(sec(x) - csc(2*x)), x) == 2*pi
+    assert periodicity(tan(sin(2*x)), x) == pi
+    raises(NotImplementedError, lambda: periodicity(exp(x), x))
+    raises(NotImplementedError, lambda: periodicity(log(x), x))
+    raises(NotImplementedError, lambda: periodicity(exp(x)**sin(x), x))
+
+
+def test_periodicity_check():
+    x = Symbol('x')
+    y = Symbol('y')
+
+    assert periodicity(tan(x), x, check=True) == pi
+    assert periodicity(sin(x) + cos(x), x, check=True) == 2*pi
```

Comment:
> Perhaps the value returned for a constant could be 0. It is not a period mathematically, but it is different from None and valid periods, and can be easily tested.

I have a small concern here.
We are asking this method (`periodicity`) to return the period of a function with respect to a variable.
And if the concerned function is constant with respect to the variable itself, raising an error would be more suitable, wouldn't it ?

For example:  Currently, we raise a `ValueError` for this function call:
```python
In []: periodicity(sin(y), x)
ValueError: The specified variable is not present in the function.
```
Similarly, would it not be a good idea to raise a `ValueError` stating that the period of a constant function is undefined in the given variable ?